poll/check.go

```diff
@@ -0,0 +1,38 @@
+package poll
+
+import (
+	"net"
+	"os"
+)
+
+// Check is a function which will be used as check for the WaitOn method.
+type Check func(t LogT) Result
+
+// FileCheck looks on filesystem and check that the path exists.
+func FileCheck(path string) Check {
```

Comment:
I imagine there may be different states of a file someone might want to poll on. I think it would be good to name this `FileExists` so that it reads well when used:

```go
poll.WaitOn(t, FileExists(filename))
```

`Exists` is also consistent with the termniology golang uses (`IsNotExist`).